uber/site_sections/preregistration.py

```diff
@@ -672,13 +672,46 @@ def not_found(self, id, message=''):
 
     def abandon_badge(self, session, id):
         attendee = session.attendee(id)
-        success_message = "Sorry you can't make it! We hope to see you next year!"
-        failure_message = "You cannot abandon your badge because you are the leader of a group."
+        if attendee.amount_paid:
+            failure_message = "Something went wrong with your refund. Please contact us at {}."\
+                .format(c.REGDESK_EMAIL)
+            new_status = c.REFUNDED_STATUS
+            page_redirect = 'repurchase'
+        else:
+            success_message = "Sorry you can't make it! We hope to see you next year!"
+            new_status = c.INVALID_STATUS
+            page_redirect = 'invalid_badge'
+            if attendee.is_group_leader:
+                failure_message = "You cannot abandon your badge because you are the leader of a group."
+            else:
+                failure_message = "You cannot abandon your badge for some reason. Please contact us at {}."\
+                    .format(c.REGDESK_EMAIL)
 
-        # a group leader may not abandon their badge
-        if attendee.is_group_leader:
+        if (not attendee.amount_paid and not attendee.can_abandon_badge)\
```

Comment:
This is kind of overkill, but I really did not want a case where someone can abandon their badge but not refund it to somehow lead to us refunding their badge anyway. We should never have a paid attendee who can abandon their badge, but if somehow that happens, I'd rather just throw them back to the page.